hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/NetUtils.java

```diff
@@ -218,6 +211,28 @@ public static InetSocketAddress createSocketAddr(String target,
     return createSocketAddrForHost(host, port);
   }
 
+  private static final Map<String, URI> URI_CACHE = new ConcurrentHashMap<>();
```

Comment:
Thanks @1996fanrui for your proposal here.
A. Not sure if this improvement could impact some other modules because {{NetUtil}} is very common util for different modules rather than DFSClient only.
B. It is better to make the scope to DFSClient only if we could based on the original description [HADOOP-17222](https://issues.apache.org/jira/browse/HADOOP-17222).
C. add evict policy for {{URI_CACHE}}?

Reply:
Hi @Hexiaoqiao , thank you very much for your suggestions.
AB. Now Cache is just URI, URI object does not contain ip, only some fixed information. The same target corresponds to the same URI. If the URI contains some variable attributes, I think the current Cache may impact other modules, and even the hdfs client will also be affected. According to my understanding, cache URI should be safe.
C. Is `evict policy` considering that the URI will change? Or consider it may take up more memory?

Reply:
> Thanks @1996fanrui for your proposal here.
> A. Not sure if this improvement could impact some other modules because {{NetUtil}} is very common util for different modules rather than DFSClient only.
> B. It is better to make the scope to DFSClient only if we could based on the original description [HADOOP-17222](https://issues.apache.org/jira/browse/HADOOP-17222).
> C. add evict policy for {{URI_CACHE}}?

Another idea: add a parameter isUseURICache to the createSocketAddr() method.
isUseURICache default is false.

Reply:
Hi @Hexiaoqiao 
DFSUtilClient.isLocalAddress also uses Cache for acceleration. It is basically similar to the maximum data size of the current patch Cache, that is, the number of DNs. The Cache of DFSUtilClient.isLocalAddress does not use the evict policy, so the current patch should not worry about taking up too much memory.
Of course, if the evict policy is added, the code will be more robust. 
If necessary, I can use guava Cache instead of Map. What do you think?

https://github.com/apache/hadoop/blob/82a750564631eb3077603bed27058006b3d66309/hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DFSUtilClient.java#L634